[WFLY-20533] Do not set the -Djava.security.manager=allow by default. Only add the property if -secmgr is passed.

jamezp opened the pull request:
This PR removes the `-Djava.security.manager=allow` argument by default and only adds it if the `-secmgr` is added.

The `ws*` scripts seem to support this, but it doesn't make sense to really do this I would think. However, I left it as I didn't want to change the behavior if the security manager is enabled for whatever reason.

This is the full version https://github.com/wildfly/wildfly-core/pull/6391. Note the way the scripts work in the tests for this PR, it's not really possible to check the JVM parameters. For that reason I was not able to validate via a test. However, this should work.

* https://issues.redhat.com/browse/WFLY-20533
* https://issues.redhat.com/browse/WFLY-20537
* https://issues.redhat.com/browse/WFLY-20538

jamezp committed:
[WFLY-20537] Upgrade WildFly Launcher to 1.0.2.Final.

https://issues.redhat.com/browse/WFLY-20537
Signed-off-by: James R. Perkins <jperkins@redhat.com>

jamezp committed:
[WFLY-20538] Upgrade WildFly Arquillian to 5.1.0.Beta11.

https://issues.redhat.com/browse/WFLY-20538
Signed-off-by: James R. Perkins <jperkins@redhat.com>

jamezp committed:
[WFLY-20533] Do not set the -Djava.security.manager=allow by default. Only add the property if -secmgr is passed.

Small fix to property destroy the processes for the script tests.

https://issues.redhat.com/browse/WFLY-20533
Signed-off-by: James R. Perkins <jperkins@redhat.com>

jamezp commented on pom.xml:
```diff
@@ -1526,17 +1525,6 @@
             </modules>
         </profile>
 
-        <profile>
-            <id>jdk18</id>
-            <activation>
-                <jdk>[18,24)</jdk>
-            </activation>
-            <properties>
-                <security.manager.props>
-                    -Djava.security.manager=allow
-                </security.manager.props>
-            </properties>
-        </profile>
```
Note, there is no need for this which is why I removed it. By default, the wildfly-launcher will add the appropriate JVM arguments. I assume this was added originally as a workaround.

bstansberry commented on ee-feature-pack/galleon-shared/src/main/resources/content/bin/appclient.bat:
```diff
@@ -83,10 +83,12 @@ if "x%JAVA_HOME%" == "x" (
+    call "!DIRNAME!common.bat" :setDefaultModularJvmOptions "!JAVA_OPTS!"
+    set "JAVA_OPTS=!JAVA_OPTS! !DEFAULT_MODULAR_JVM_OPTIONS!"
+    setlocal DisableDelayedExpansion
+)
```
This is not about SM. It's about --add-opens etc and should not be changed as part of this JIRA.  If something needs changing it's another JIRA.

jamezp replied:
Good catch on all these. I clearly scanned for the wrong function name.

bstansberry commented on ee-feature-pack/galleon-shared/src/main/resources/content/bin/jdr.bat:
```diff
@@ -50,12 +50,6 @@ if "x%JAVA_HOME%" == "x" (
-setlocal EnableDelayedExpansion
-call "!DIRNAME!common.bat" :setDefaultModularJvmOptions "!JAVA_OPTS!"
-set "JAVA_OPTS=!JAVA_OPTS! !DEFAULT_MODULAR_JVM_OPTIONS!"
-setlocal DisableDelayedExpansion
```
Not about SM; shouldn't be changed.

bstansberry commented on ee-feature-pack/galleon-shared/src/main/resources/content/bin/jdr.sh:
```diff
@@ -57,10 +57,6 @@ if [ "x$JAVA" = "x" ]; then
 
-# Set default modular JVM options
-setDefaultModularJvmOptions $JAVA_OPTS
-JAVA_OPTS="$JAVA_OPTS $DEFAULT_MODULAR_JVM_OPTIONS"
```
Not about SM; shouldn't be changed.

bstansberry commented on ee-feature-pack/galleon-shared/src/main/resources/content/bin/wsconsume.bat:
```diff
@@ -49,10 +49,12 @@ if "x%JAVA_HOME%" == "x" (
+    call "!DIRNAME!common.bat" :setDefaultModularJvmOptions "!JAVA_OPTS!"
+    set "JAVA_OPTS=!JAVA_OPTS! !DEFAULT_MODULAR_JVM_OPTIONS!"
+    setlocal DisableDelayedExpansion
+)
```
Not about SM; shouldn't be changed.

bstansberry commented on ee-feature-pack/galleon-shared/src/main/resources/content/bin/wsconsume.sh:
```diff
@@ -63,8 +63,10 @@ if [ "x$JAVA" = "x" ]; then
+if [ "$SECMGR" = "true" ]; then
+    setDefaultModularJvmOptions $JAVA_OPTS
+    JAVA_OPTS="$JAVA_OPTS $DEFAULT_MODULAR_JVM_OPTIONS"
+fi
```
Not about SM; shouldn't be changed.

bstansberry commented on ee-feature-pack/galleon-shared/src/main/resources/content/bin/wsprovide.bat:
```diff
@@ -47,12 +47,12 @@ if "x%JAVA_HOME%" == "x" (
+    call "!DIRNAME!common.bat" :setDefaultModularJvmOptions "!JAVA_OPTS!"
+    set "JAVA_OPTS=!JAVA_OPTS! !DEFAULT_MODULAR_JVM_OPTIONS!"
+    setlocal DisableDelayedExpansion
+)
```
Not about SM; shouldn't be changed.

bstansberry commented on ee-feature-pack/galleon-shared/src/main/resources/content/bin/wsprovide.sh:
```diff
@@ -63,8 +63,10 @@ if [ "x$JAVA" = "x" ]; then
+if [ "$SECMGR" = "true" ]; then
+    setDefaultModularJvmOptions $JAVA_OPTS
+    JAVA_OPTS="$JAVA_OPTS $DEFAULT_MODULAR_JVM_OPTIONS"
+fi
```
Not about SM; shouldn't be changed.

bstansberry commented:
@jamezp The SE 21 failures look related, perhaps from changes in wildfly-launcher?

jamezp commented:
They do look related indeed. I'll have a look as soon as I can.

jamezp commented:
> Looks like it's caused by the unit test design itself. Maybe the failed ones should only run if the sec manager is enabled, or this specific test case itself should be run with `-Djava.security.manager=allow`

Yes, that was it. I had removed the argument from the main POM and now just added it back to the `naming/pom.xml`

jamezp commented:
Not sure why it's not updating, but CI is running on https://ci.wildfly.org/project/WF_PullRequest?branch=pull%2F18868&mode=builds

jamezp commented:
The last failure is unrelated.

bstansberry commented:
Thanks @jamezp
